rllib/examples/rl_modules/classes/action_masking_rlm.py

```diff
@@ -1,121 +1,133 @@
-    logits = outputs[SampleBatch.ACTION_DIST_INPUTS]
-    # Convert action_mask into a [0.0 || -inf]-type mask.
-    inf_mask = torch.clamp(torch.log(action_mask), min=FLOAT_MIN)
-    masked_logits = logits + inf_mask
-
-    # Replace original values with masked values.
-    outputs[SampleBatch.ACTION_DIST_INPUTS] = masked_logits
-
-    return outputs
-
-
-def mask_forward_fn_tf(forward_fn, batch, **kwargs):
-    _check_batch(batch)
-
-    # Extract the available actions tensor from the observation.
-    action_mask = batch[SampleBatch.OBS]["action_mask"]
-
-    # Modify the incoming batch so that the default models can compute logits and
-    # values as usual.
-    batch[SampleBatch.OBS] = batch[SampleBatch.OBS]["observations"]
-
-    outputs = forward_fn(batch, **kwargs)
-
-    # Mask logits
-    logits = outputs[SampleBatch.ACTION_DIST_INPUTS]
-    # Convert action_mask into a [0.0 || -inf]-type mask.
-    inf_mask = tf.maximum(tf.math.log(action_mask), tf.float32.min)
-    masked_logits = logits + inf_mask
+        # While the environment holds an observation space that contains, both,
+        # the action mask and the original observation space, the 'RLModule'
+        # receives only the `"observation"` element of the space, but not the
+        # action mask.
+        self.observation_space_with_mask = config.observation_space
+        config.observation_space = config.observation_space["observations"]
 
-    # Replace original values with masked values.
-    outputs[SampleBatch.ACTION_DIST_INPUTS] = masked_logits
+        # Keeps track if observation specs have been checked already.
+        self._checked_observations = False
 
-    return outputs
+        # The PPORLModule, in its constructor will build networks for the original
+        # observation space (i.e. without the action mask).
+        super().__init__(config)
 
 
-def _check_batch(batch):
-    """Check whether the batch contains the required keys."""
-    if "action_mask" not in batch[SampleBatch.OBS]:
-        raise ValueError(
-            "Action mask not found in observation. This model requires "
-            "the environment to provide observations that include an "
-            "action mask (i.e. an observation space of the Dict space "
-            "type that looks as follows: \n"
-            "{'action_mask': Box(0.0, 1.0, shape=(self.action_space.n,)),"
-            "'observations': <observation_space>}"
-        )
-    if "observations" not in batch[SampleBatch.OBS]:
-        raise ValueError(
-            "Observations not found in observation.This model requires "
-            "the environment to provide observations that include a "
-            " (i.e. an observation space of the Dict space "
-            "type that looks as follows: \n"
-            "{'action_mask': Box(0.0, 1.0, shape=(self.action_space.n,)),"
-            "'observations': <observation_space>}"
-        )
+class ActionMaskTorchRLM(ActionMaskRLM, PPOTorchRLModule):
+    def setup(self):
```

Comment:
nit: add `override` decorators to all methods.

Reply:
Good catch!